### DISCUSSION TITLE
Limit Arm Motion for Large Characters

### DISCUSSION THREAD

**Discussion by matttidridge** [endink/Mediapipe4u-plugin#137] [Q&A]
I an trying to limit (or clamp) the arm motion in the Animation Blueprint so that the arms do not clip inside the body of a larger character. 

Is there a good way to get the arm rotations and limit them?

**Comment by endink** [comment]
The limited rotation of joints is in my roadmap for a long time, and it will take a lot of code to implement this feature for different skeletons in FK mode.  

I think transfer  FK to control rig IK is a better choice. In ControlRig , every user can limit bone rotation easily , I have done this （let M4U support control rig）in another project.

So, I will bring control rig support into M4U asap. But right now, I can't give a exact timeline.

**Reply by matttidridge** [reply]
Appreciate the quick response. Thank you for thinking about this feature.

There is no way currently to get the arm bone rotation from the Pose node so I can perform a clamp on it in Blueprints is there?

**Comment by endink** [comment]
Try latest version, use `Control Rig From Pose` node in your animation blueprint, so that you can make bone limitation in ControlRig blueprint.
